akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/SlotState.scala

```diff
@@ -191,10 +191,11 @@ private[pool] object SlotState {
   private[pool] case object ToBeClosed extends ShouldCloseConnectionState(None)
   private[pool] case class Failed(cause: Throwable) extends ShouldCloseConnectionState(Some(cause))
 
-  private[pool] case object Idle extends ConnectedState with IdleState {
+  private[pool] final case class Idle(override val stateTimeout: Duration) extends ConnectedState with IdleState {
```

Comment:
Let's introduce a member called "maxConnectionIdleTimeout` to make it more explicit what this particular timeout is about.

```suggestion
  private[pool] final case class Idle(maxConnectionIdleTimeout: Duration) extends ConnectedState with IdleState {
    override def stateTimeout: Duration = maxConnectionIdleTimeout
```